sdks/java/extensions/sql/datacatalog/src/main/java/org/apache/beam/sdk/extensions/sql/meta/provider/datacatalog/PubsubTableFactory.java

```diff
@@ -20,22 +20,32 @@
 import com.alibaba.fastjson.JSONObject;
 import com.google.cloud.datacatalog.Entry;
 import java.net.URI;
+import java.util.Optional;
 import java.util.regex.Matcher;
 import java.util.regex.Pattern;
 import org.apache.beam.sdk.extensions.sql.meta.Table;
+import org.apache.beam.sdk.extensions.sql.meta.Table.Builder;
 
-/** Utils to extract Pubsub-specific entry information. */
-class PubsubUtils {
+/** {@link TableFactory} that understands Data Catalog Pubsub entries. */
+class PubsubTableFactory implements TableFactory {
+
+  private static final String PUBSUB_API = "pubsub.googleapis.com";
 
   private static final Pattern PS_PATH_PATTERN =
       Pattern.compile("/projects/(?<PROJECT>[^/]+)/topics/(?<TOPIC>[^/]+)");
 
-  static Table.Builder tableBuilder(Entry entry) {
-    return Table.builder()
-        .location(getLocation(entry))
-        .properties(new JSONObject())
-        .type("pubsub")
-        .comment("");
+  @Override
+  public Optional<Builder> tableBuilder(Entry entry) {
+    if (URI.create(entry.getLinkedResource()).getAuthority().toLowerCase().equals(PUBSUB_API)) {
+      return Optional.of(
+          Table.builder()
+              .location(getLocation(entry))
+              .properties(new JSONObject())
+              .type("pubsub")
+              .comment(""));
+    } else {
```

Comment:
nit: Unneeded else. This might be clearer if the `if` bailed out.

Reply:
Done.